### DISCUSSION TITLE
Understanding more about custom payment processing specifically for subscription

### DISCUSSION THREAD

**Discussion by y3owk1n** [swellstores/community#51] [Help] [answered]
Hello everyone, I am having this scenario that I need to handle both (one off purchase and subscription). However the hosted checkout are not viable for me mainly because of certain local payment gateway (eg FPX, Atome) here in Malaysia are not available natively in Swell.

I've successfully implemented a custom checkout for one off payments with all of the desired local payment gateways. However, when it comes to subscription products, how exactly is the best way to handle it?

I understand that all of the local payments mentioned above are best suited for one off payments. When it comes to subscription orders, do i need to use specific gateway (eg Stripe) to handle the payment? 

Referencing the [doc](https://developers.swell.is/frontend-api/payments), it seems like I can freely implement any gateway as per my liking, but how do I ensure the recurring charges accordingly?

Attach my current store for better references.
Current staging site: https://uat.herlianaturals.com

**Comment by ericingram** [comment, marked as answer]
Beautiful site! 🙌 

You can handle subscription and order payments for any gateway that isn't natively supported, using a "Manual payment method" (Settings > Payments) along with a webhook.

If you have the "Charge automatically" setting enabled (Settings > Orders/Subscriptions), then the system will automatically create a payment when the order OR subscription is to be charged, which can then fire a webhook event `payment.created` for you to ping your own payment gateway. The webhook handler should update the payment record with details about success or failure of the transaction.

If you didn't want to enable "Charge automatically", you can do something similar by listening to webhook events `order.submitted` (for orders) and `invoice.created` (for subscriptions).

**Reply by y3owk1n** [reply]
@ericingram Thanks! I'm understanding the subscription logic more now. The above are referring to non supported gateways, which I will need to handle the payments manually. 

My another questions is, if I would like to implement say using the Stripe integration out of the box (refer to the following image), but on my own custom checkout flow rather than the hosted checkout, is there any resources that you can perhaps point me to?
![image](https://user-images.githubusercontent.com/62775956/209025799-6276f137-fd30-49d6-beb9-ff8616a1b3c2.png)

The ideal flow that I am hoping to achieve is having swell to handle the recurring charges automatically without more efforts. Something as below:

1. Tokenize a card and attach it to the subscription order.
2. Swell to handle the recurring charges automatically until it gets cancelled by the customer.

**Reply by sags95** [reply]
Hey @y3owk1n , I think you would find [this](https://developers.swell.is/guides/render-stripe-elements-with-react) guide useful, as it describes how to setup a custom checkout with Stripe. For carts with subscription purchase_options, Swell will automatically use the tokenized card for future charges.

**Reply by y3owk1n** [reply]
@sags95 Perfect, i am all clear now and thank you so much for the help here!
